GVFS/GVFS.Common/Git/GitAuthentication.cs

```diff
@@ -71,39 +77,28 @@ public bool TryRefreshCredentials(ITracer tracer, out string errorMessage)
 
         public bool TryGetCredentials(ITracer tracer, out string gitAuthString, out string errorMessage)
         {
+            if (!this.isInitialized)
+            {
+                throw new InvalidOperationException("This auth instance must be initialized before it can be used");
+            }
+

```

Comment:
In theory, it's possible for anonymous access to be allowed initially, and then disallowed (because someone changed the setting on the server). That's not a tested scenario though, so I'd be fine with disallowing it right now.